resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java

```diff
@@ -185,6 +187,10 @@ private void stateTransition(State newState, Function<CircuitBreakerState, Circu
         });
         if (previousState.getState() != newState) {
             publishStateTransitionEvent(StateTransition.transitionBetween(previousState.getState(), newState));
+
+            if (circuitBreakerConfig.getEnableAutomaticTransitionFromOpenToHalfOpen() && newState.equals(OPEN)) {
+                OpenToHalfOpenAutoTransitioner.scheduleAutoTransitionToHalfOpen(this);
```

Comment:
OpenToHalfOpenAutoTransitioner.scheduleAutoTransitionToHalfOpen() should be invoked in the Constructor of `OpenState`.